dom/test/uix/dom/server_test.cljc

```diff
@@ -403,3 +403,14 @@
      (let [_ (render ($ error-boundary-catches))]
        ;; Tests that did-catch does run
        (is (.contains @*error-state "error-boundary")))))
+
+(defui username [{:keys [username]}]
+  ($ :p {:dangerouslySetInnerHTML {:__html username}}))
+
+(defui bad-username [_]
+  ($ :p {:dangerouslySetInnerHTML {:__not-html "foo"}}))
+
+#?(:clj
+   (deftest nil-dangerously-set-inner-html
+     (is (= (render ($ username {:username nil})) "<p></p>"))
+     (is (thrown? Exception (render ($ bad-username {:foo 1}))))))
```

Comment:
shouldn't this test fail since the rendering code checks for presence of a key?

this is also how React SSR works, they only check for presence of a key https://codesandbox.io/s/suspicious-gould-f3jxg9?file=/src/index.js

Reply:
Which test exactly? The first or second `is` check?

Reply:
uh, sorry, I miss read the change, looks good! :D

Reply:
you made me super confused :D had to re-read the code a couple times!